### PR TITLE
🛠Keep the line info of config schema validation

### PR DESCRIPTION
**Pull request by live1206**

### CODE REVIEW

File: src/docfx/config/ConfigLoader.cs [hunk 1]
```diff
@@ -154,8 +157,11 @@ private static (List<Error>, JObject) ExtendConfigs(JObject config, string docse
                 }
             }
 
-            JsonUtility.Merge(result, config);
-            return (errors, result);
+            // to keep the line info of config
```
Comment (OsmondJiang) [inline src/docfx/config/ConfigLoader.cs]: may we extract this as a method?

Comment (live1206) [reply src/docfx/config/ConfigLoader.cs]: Thought it was not a good idea to put the workaround into lib, but maybe it's ok

File: src/docfx/lib/JsonUtility.cs [hunk 2]
```diff
@@ -268,6 +268,17 @@ public static void Merge(JObject container, JObject overwrite)
             container.Merge(overwrite, s_mergeSettings);
         }
 
+        /// <summary>
+        /// Merge overwrite into container
+        /// And keep the line info of properties in overwrite which are not in container
+        /// </summary>
+        public static void MergeWithLineInfo(JObject container, JObject overwrite)
+        {
+            var original = overwrite.DeepClone() as JObject;
```
Comment (OsmondJiang) [inline src/docfx/lib/JsonUtility.cs]: how about make it as default behavior?

Comment (live1206) [reply src/docfx/lib/JsonUtility.cs]: sounds good

Comment (live1206) [reply src/docfx/lib/JsonUtility.cs]: Actually, it depends if we need this workaround

Comment (OsmondJiang) [reply src/docfx/lib/JsonUtility.cs]: move this method to configLoader as a private method

**Pull request merged**